### PR TITLE
**[FIX]** Fixes issue #705

### PR DESCRIPTION
**Pull request by Diptanshu8**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used CCExtractor.
- [x] I have used CCExtractor just a couple of times.
- [ ] I absolutely love CCExtractor, but have not contributed previously.
- [ ] I am an active  #contributor to CCExtractor.

---

{pull request content here}
fixes points 1,2,4 and additional `-mklang` option mentioned in #705

### CODE REVIEW

File: src/lib_ccx/matroska.c [hunk 1]
```diff
@@ -738,6 +743,7 @@ char* generate_filename_from_track(struct matroska_ctx* mkv_ctx, struct matroska
 
 char* ass_ssa_sentence_erase_read_order(char* text)
 {
+    return text;
```
Comment (Izaron) [inline src/lib_ccx/matroska.c]: Remove this, it was just for debugging

File: src/lib_ccx/matroska.c [hunk 2]
```diff
@@ -814,6 +823,9 @@ void save_sub_track(struct matroska_ctx* mkv_ctx, struct matroska_sub_track* tra
             write(desc, timestamp_end, strlen(timestamp_start));
             write(desc, ",", 1);
             char* text = ass_ssa_sentence_erase_read_order(sentence->text);
+            while(text[0]=='\n' || text[0]=='\N'){
```
Comment (Izaron) [inline src/lib_ccx/matroska.c]: ASS/SSA line ending is two chars. Check that text[0] is '\' and text[1] is 'n' or 'N'

**Comment by Diptanshu8** [comment]
@Izaron The quick fixes have been made. Kindly, check.

**Comment by Izaron** [comment]
Well, seems you don't have tested it well as seems to me...
All I require is thorough testing of 3 examples. It isn't so hard.

So, I tested on my own and got list of bugs:
1) Compilation error - you should change signature of changed function https://github.com/Diptanshu8/ccextractor/blob/b723378eb26f873a62dac721e73807247fa94203/src/lib_ccx/matroska.h#L251
2) https://github.com/Diptanshu8/ccextractor/blob/b723378eb26f873a62dac721e73807247fa94203/src/lib_ccx/matroska.c#L942 - you should do `realloc(sizeof(struct matroska_sub_track**) * track_cnt)` for every new track, and then `mkv_ctx->sub_tracks[mkv_ctx->sub_tracks_count] = sub_track;`, othwerwise all will get segfault
3) 4th point SRT - in vikings.mkv there are `"\r\n\r\n\r\n\r\nСпасибо, брат."`, not just `"\n\n\n\nСпасибо, брат."` Why you don't checked this '\r\n' for line endings?
4) 4th point ASS/SSA - in vikings.mkv there are `"V2,,0000,0000,0000,,\\NСпасибо, брат."`, this should be `"V2,,0000,0000,0000,,Спасибо, брат."` and so on
5) https://github.com/Diptanshu8/ccextractor/blob/b723378eb26f873a62dac721e73807247fa94203/src/lib_ccx/matroska.c#L805 - I would like have only `size++` and then `write(desc, sentence->text + size, sentence->text_size - size);`, because you should have segfault at this line https://github.com/Diptanshu8/ccextractor/blob/b723378eb26f873a62dac721e73807247fa94203/src/lib_ccx/matroska.c#L845 at the current version
6) https://github.com/Diptanshu8/ccextractor/blob/b723378eb26f873a62dac721e73807247fa94203/src/lib_ccx/matroska.c#L766 - you should do "size = 0" for every new sentence. In the current version, if you have "size = 6", size never will be less than 6, and every next sentences never will have first 6 chars printed.

Please test your code on this 3 samples and use good IDE for working on (CLion/Visual Studio)

**Pull request merged**